### PR TITLE
Prepare for release

### PR DESCRIPTION
**Pull request by alexhung**
- Update code to use shared module
- Add documentation
- Update GoReleaser to use Go 1.18
- Add telemetry
- Update make file and test scripts
- Add provider test (other acceptance tests need to wait until we figure out how to install Pipeline locally (if at all possible))

### CODE REVIEW

File: pkg/pipeline/provider.go [hunk 1]
```diff
@@ -3,27 +3,26 @@ package pipeline
...
 
 func Provider() *schema.Provider {
 	p := &schema.Provider{
 		Schema: map[string]*schema.Schema{
 			"url": {
 				Type:         schema.TypeString,
 				Required:     true,
-				DefaultFunc:  schema.MultiEnvDefaultFunc([]string{"PIPELINES_URL", "JFROG_URL"}, "http://localhost:8081"),
+				DefaultFunc:  schema.MultiEnvDefaultFunc([]string{"PIPELINES_URL", "JFROG_URL"}, "http://localhost:8082"),
 				ValidateFunc: validation.IsURLWithHTTPorHTTPS,
```
Comment (chb0github) [inline pkg/pipeline/provider.go:24]: Please use new style validation

File: pkg/pipeline/resource_pipeline_node_pool.go [hunk 2]
```diff
@@ -85,24 +87,24 @@ func pipelineNodePoolResource() *schema.Resource {
...
-			IsOnDemand:             d.getBool("is_on_demand"),
-			Architecture:           d.getString("architecture"),
-			OperatingSystem:        d.getString("operating_system"),
-			NodeIdleIntervalInMins: d.getInt("node_idle_interval_in_mins"),
-			Environments:           d.getList("environments"),
+			ProjectId:              d.GetInt("project_id", false),
+			Name:                   d.GetString("name", false),
+			NumberOfNodes:          d.GetInt("number_of_nodes", false),
+			IsOnDemand:             d.GetBool("is_on_demand", false),
+			Architecture:           d.GetString("architecture", false),
```
Comment (chb0github) [inline pkg/pipeline/resource_pipeline_node_pool.go:97]: Is there strict validation on this?

File: pkg/pipeline/resource_pipeline_project_integration.go [hunk 3]
```diff
@@ -158,24 +160,24 @@ func pipelineProjectIntegrationResource() *schema.Resource {
...
+		d := &util.ResourceData{data}
 
 		projectIntegration := ProjectIntegration{
-			Name:                  d.getString("name"),
-			ProjectId:             d.getInt("project_id"),
-			MasterIntegrationId:   d.getInt("master_integration_id"),
-			MasterIntegrationName: d.getString("master_integration_name"),
-			Environments:          d.getList("environments"),
-			IsInternal:            d.getBool("is_internal"),
+			Name:                  d.GetString("name", false),
```
Comment (chb0github) [inline pkg/pipeline/resource_pipeline_project_integration.go:166]: This is almost always false. I wish I knew what this function was meant to do

File: pkg/pipeline/resource_pipeline_project_integration.go [hunk 4]
```diff
@@ -158,24 +160,24 @@ func pipelineProjectIntegrationResource() *schema.Resource {
...
+			Environments:          d.GetList("environments"),
+			IsInternal:            d.GetBool("is_internal", false),
 			Project:               unpackProject(d, "project"),
 			FormJSONValues:        unpackFormJSONValues(d, "form_json_values"),
 		}
 		return projectIntegration, nil
 	}
 
-	var packProjectIntegration = func(d *schema.ResourceData, projectIntegration ProjectIntegration) diag.Diagnostics {
+	var packProjectIntegration = func(ctx context.Context, d *schema.ResourceData, projectIntegration ProjectIntegration) diag.Diagnostics {
```
Comment (chb0github) [inline pkg/pipeline/resource_pipeline_project_integration.go:178]: Universal pack won't work?

File: pkg/pipeline/provider.go [hunk 5]
```diff
@@ -47,6 +46,7 @@ func Provider() *schema.Provider {
 			"pipeline_node_pool":           pipelineNodePoolResource(),
 			"pipeline_node":                pipelineNodeResource(),
 		},
+
```
Comment (chb0github) [inline pkg/pipeline/provider.go:49]: telemetry?

**Review by chb0github — COMMENTED** [review]
For an initial release, maybe this works - but some things seem missing

File: pkg/pipeline/provider.go (hunk 5 above)
Comment (alexhung) [reply pkg/pipeline/provider.go:49]: Forgot about this one.

File: pkg/pipeline/resource_pipeline_project_integration.go (hunk 4 above)
Comment (alexhung) [reply pkg/pipeline/resource_pipeline_project_integration.go:178]: I'm not planning to update the code in this PR.

File: pkg/pipeline/resource_pipeline_node_pool.go (hunk 2 above)
Comment (alexhung) [reply pkg/pipeline/resource_pipeline_node_pool.go:97]: I'm not planning to update the code in this PR.

File: pkg/pipeline/provider.go (hunk 1 above)
Comment (alexhung) [reply pkg/pipeline/provider.go:24]: I'm not planning to update the code in this PR.

**Review by chb0github — APPROVED** [review]

File: sample.tf [hunk 6]
```diff
@@ -0,0 +1,76 @@
+terraform {
```
Comment (chb0github) [inline sample.tf:1]: You're definitely going to want to include an example of how it integrates with the other providers

File: sample.tf [hunk 7]
```diff
@@ -2,7 +2,15 @@ terraform {
...
-      version = "0.1.0"
+      version = "0.1.1"
+    }
+    project = {
+      source  = "registry.terraform.io/jfrog/project"
+      version = "1.1.0"
+    }
+    artifactory = {
+      source  = "registry.terraform.io/jfrog/artifactory"
+      version = "6.7.1"
```
Comment (danielmkn) [inline sample.tf:13]: do we want to have other providers here? We don't do this for other projects.

**Review by maheshjfrog — APPROVED** [review]

**Review by danielmkn — APPROVED** [review]

File: sample.tf (hunk 7 above)
Comment (alexhung) [reply sample.tf:13]: We don't currently but we should do the same for Xray and Project providers as well

**Pull request merged**